jbi/router.py

```diff
@@ -75,6 +76,20 @@ async def bugzilla_webhook(
     return await execute_or_queue(webhook_request, queue, actions)
 
 
+@router.get(
+    "/dl_queue/",
+    dependencies=[Depends(api_key_auth)],
+)
+async def inspect_dl_queue(queue: Annotated[DeadLetterQueue, Depends(get_dl_queue)]):
+    """API for viewing queue content"""
+    bugs = await queue.retrieve()
+    results = defaultdict(list)
+    for bug_id, items in bugs.items():
+        async for item in items:
```

Comment:
If we're going to read each item here, we need to account for the possibility that the webhook payload somehow doesn't match our schema and throws a validation exception. That's why the current code to retrieve all of the items in the queue only retrieves the identifiers

Reply:
I don't understand

Do you mean that we would want to be able to inspect the invalid items in queue via this endpoint?

Reply:
I think my confusion comes from the fact that we still return 422 when incoming payloads are invalid. Or am I wrong?

Reply:
I opened #967 and #968